src/main/java/appeng/bootstrap/FeatureFactory.java

```diff
@@ -123,39 +127,43 @@ public FeatureFactory features( AEFeature... features )
 		return new FeatureFactory( this, features );
 	}
 
-	void addBootstrapComponent( IBootstrapComponent component )
+	public <T extends IBootstrapComponent> void addBootstrapComponent( T component )
 	{
-		this.bootstrapComponents.add( component );
+		Arrays.stream( component.getClass().getInterfaces() )
+				.filter( i -> IBootstrapComponent.class.isAssignableFrom( i ) )
+				.forEach( i -> this.addBootstrapComponent( (Class<? extends IBootstrapComponent>) i, component ) );
 	}
 
-	void addPreInit( IPreInitComponent component )
+	private <T extends IBootstrapComponent> void addBootstrapComponent( Class<? extends IBootstrapComponent> eventType, T component )
 	{
-		this.bootstrapComponents.add( component );
-	}
-
-	void addInit( IInitComponent component )
-	{
-		this.bootstrapComponents.add( component );
-	}
-
-	void addModelReg( IModelRegistrationComponent component )
-	{
-		this.bootstrapComponents.add( component );
+		List<IBootstrapComponent> eventList = bootstrapComponents.get( eventType );
+		if( eventList == null )
+		{
+			eventList = new ArrayList<>();
+			bootstrapComponents.put( eventType, eventList );
+		}
+		eventList.add( component );
 	}
 
-	void addPostInit( IPostInitComponent component )
+	@SideOnly( Side.CLIENT )
+	void addBuiltInModel( String path, IModel model )
 	{
-		this.bootstrapComponents.add( component );
+		this.builtInModelComponent.addModel( path, model );
 	}
 
 	@SideOnly( Side.CLIENT )
-	void addBuiltInModel( String path, IModel model )
+	void addModelOverride( String resourcePath, BiFunction<ModelResourceLocation, IBakedModel, IBakedModel> customizer )
 	{
-		this.builtInModelComponent.addModel( path, model );
+		this.modelOverrideComponent.addOverride( resourcePath, customizer );
 	}
 
-	public List<IBootstrapComponent> getBootstrapComponents()
+	public <T extends IBootstrapComponent> Iterator<T> getBootstrapComponents( Class<T> eventType )
 	{
-		return this.bootstrapComponents;
+		List<IBootstrapComponent> ret = this.bootstrapComponents.get( eventType );
```

Comment:
Could use `getOrDefault(eventType , Collections.emptyList())` to reduce the LoCs a bit.